### PR TITLE
Watermark Plugin

### PR DESCRIPTION
**Pull request by abdul**
This plugin allows adding watermark to images.

### CODE REVIEW

File: sigal/plugins/watermark.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+
+"""Plugin which adds a watermark to the image.
+
+Settings:
+
+- ``watermark``: path to the watermark image.
+- ``watermark_position``: the watermark position (scale or tile)
```
Comment (saimn) [inline sigal/plugins/watermark.py]: From the code it seems that it is also possible to specify the position with a tuple which is given to http://pillow.readthedocs.org/reference/Image.html#PIL.Image.Image.paste
Could you also document this behavior ?

File: sigal/plugins/watermark.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+def add_watermark(img, settings=None):
+    logger.debug('Adding watermark to %r', img)
+    mark = Image.open(settings['watermark'])
+    position = 'scale'
+    opacity = 1
+    if settings['watermark_position']:
+        position = settings['watermark_position']
+    if settings['watermark_opacity']:
+        opacity = settings["watermark_opacity"]
```
Comment (saimn) [inline sigal/plugins/watermark.py]: You could use `settings("watermark_opacity", 1)` to simplify the code, the same for `watermark_position`.

File: sigal/plugins/watermark.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        mark = mark.resize((w, h))
+        layer.paste(mark, ((im.size[0] - w) / 2, (im.size[1] - h) / 2))
+    else:
+        layer.paste(mark, position)
+    # composite the watermark with the layer
+    return Image.composite(layer, im, layer)
+
+def add_watermark(img, settings=None):
+    logger.debug('Adding watermark to %r', img)
+    mark = Image.open(settings['watermark'])
```
Comment (saimn) [inline sigal/plugins/watermark.py]: The path here is relative to the cwd. To allow running sigal from another directory, and be consistent with other path settings, could you add the `watermark` setting here https://github.com/saimn/sigal/blob/master/sigal/settings.py#L126 to make it relative to the settings file ?

**Comment by saimn** [comment]
Looks good (with some minor comments), thanks for the contribution !

**Comment by abdul** [comment]
I would make these changes, and update the PR. Thanks for the feedback.

**Comment by abdul** [comment]
@saimn I have update the code, please review. Thank you.

**Comment by saimn** [comment]
Thanks !

**Comment by abdul** [comment]
Thank you.

On Friday, March 6, 2015, Simon Conseil <notifications@github.com> wrote:

> Thanks !
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/saimn/sigal/pull/148#issuecomment-77564692>.
>


-- 
Sent from Gmail Mobile

**Pull request merged**